src/exception/exception-api.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { Request, Response } from 'express';
+import { HasFullWriteAccess } from '../decorators';
+import { loggerFor } from '../logger';
+import { ControllerUtils, TsvUtils } from '../utils';
+import * as exceptionService from './exception-service';
+
+const L = loggerFor(__filename);
+
+enum ProgramExceptionErrorMessage {
+  TSV_PARSING_FAILED = `This file is formatted incorrectly`,
+}
+
+class ExceptionController {
+  @HasFullWriteAccess()
+  async createProgramException(req: Request, res: Response) {
+    if (!isValidCreateBody(req, res)) {
+      return false;
+    }
+
+    const programId = req.params.programId;
```

Comment:
`programId` has type `any`. We need to validate that `req.params.programId` was provided and is a string.
```suggestion
    const programId = req.params.programId;
    if (!programId && programId === 'string') {
      ControllerUtils.unableToProcess(res, `Invalid ProgramID provided`);
    }
```

Reply:
the programId comes from the endpoint url, so no programId is just not an endpoint
I will check if express has all params as string

Reply:
oh fair enough. if its a path param then it has to be a string. i'm fine with that.